sys/net/gnrc/sock/udp/gnrc_sock_udp.c

```diff
@@ -174,6 +177,29 @@ int sock_udp_get_remote(sock_udp_t *sock, sock_udp_ep_t *remote)
     return 0;
 }
 
+bool sock_udp_ep_equal(const sock_udp_ep_t *a, const sock_udp_ep_t *b)
```

Comment:
Will do, but first please repair the `sock_util` module. At least on my host I get wild compile errors, as `sock_util` is using the host systems `arpa/inet.h` include, which leads to quite some clashes with RIOT internal includes... Don't really have the time nor will to look into this.

Reply:
This is fixed as a piggy-back to https://github.com/RIOT-OS/RIOT/pull/8646. As the review of that is stalled, I will cherry-pick it out ASAP.

Reply:
(as a workaround [or confirmation that that fix works] try including `posix` into your example application)

Reply:
no, that is not an option. I will not include posix into my app just because some small helper util package forces me to do that :-)

Reply:
Not even to test if my proposed fix (I will provide tomorrow when I have a keyboard) would work?

Reply:
Fix in https://github.com/RIOT-OS/RIOT/pull/9498

Reply:
works (although I don't really like the idea of pulling in the `posix` module, just because I'd like to compare two sock UDP endpoints...). But nonetheless, it works and I guess I have to live with that for now.

Reply:
That's a battle you have to fight with @kaspar030, who insists using `inet_pton()`/`inet_ntop()` instead of the `ipv6_addr_to_str()`/`ipv6_addr_from_str()` ;-).